server/server.go

```diff
@@ -104,6 +105,13 @@ func randomBuf(size int) []byte {
 	return buf
 }
 
+func (s *Server) getRealRemoteAddr(conn net.Conn) net.Addr {
```

Comment:
Ditto. Don't use `real` since `real` is not obvious. Don't name it `get` since when there is a `ppDecoder` it is no longer idempotent.